packages/backend/src/services/UserService.ts

```diff
@@ -196,7 +203,64 @@ export class UserService {
             });
             return sessionUser;
         }
-        return undefined;
+
+        // Create user
+        return this.createUserWithOpenId(openIdUser);
+    }
+
+    async createUserWithOpenId(openIdUser: OpenIdUser): Promise<SessionUser> {
+        if (!(await this.organizationModel.hasOrgs())) {
+            const user = await this.registerInitialUser(openIdUser);
+            return this.userModel.findSessionUserByUUID(user.userUuid);
+        }
+        throw new AuthorizationError(
+            'Can not create user in existing organization',
+        );
+    }
+
+    async completeUserSetup(
+        user: SessionUser,
+        {
+            organizationName,
+            jobTitle,
+            isTrackingAnonymized,
+            isMarketingOptedIn,
+        }: CompleteUserArgs,
+    ): Promise<LightdashUser> {
+        if (organizationName) {
+            await this.organizationModel.update(user.organizationUuid, {
+                organizationName,
+            });
+            analytics.track({
+                userId: user.userUuid,
+                event: 'organization.updated',
```

Comment:
Not sure we should fire this event here, as the user hasn't actually opted to do an `organization.updated` action?

Reply:
@TuringLovesDeathMetal asked for this when we chat about it. Would we rather have a separate event ? `organisation.completed`

Reply:
^ proposal in this comment https://github.com/lightdash/lightdash/pull/1181#discussion_r790625577